### PR TITLE
Milestone 2: User interface

### PR DESCRIPTION
**Pull request by dansantander**
This Milestone complies with the following requirements: 

Think about steps needed to play the game:
what do you need from player?
what do you need to display for player?

Display prompts for users, collect user input and only mock the basic flow of the game.

Make sure that you can run your game with your executable file and play with the basic flow.

The final point of this milestone is to make sure that you thought about all actions needed to play Tic Tac Toe. **It does not need to work yet** - this will be done in the next milestone.

### CODE REVIEW

**Review by pelzolga123 — CHANGES_REQUESTED** [review]
Project requires changes.

**Comment by pelzolga123** [comment]
## Hi Team!  
###  Good job so far :+1: 
In this milestone not everything must work, the point is to make sure you thought about everything. Points listed below are missing:
 
* [ ] Game should ask player to select from available moves.
* [ ] Game should inform player if selected move is invalid.
* [ ] Game should display board after player move.

### Please kindly fix the above issues and send another code review request.

If you have any questions or need help, you can message me on slack (@ Olga Pelts). *Happy coding*

File: bin/main.rb [hunk 1]
```diff
@@ -1,2 +1,104 @@
 #!/usr/bin/env ruby
-puts 'Hello World'
+public
```
Comment (suretrust) [inline bin/main.rb:47]: This doesn't need to be defined explicitly, ruby already knows.

File: bin/main.rb [hunk 2]
```diff
@@ -1,2 +1,104 @@
...
+puts '###BOARD APPEARS HERE###'
+puts '-Players take turns to choose a cell, the first one to make a line of symbols wins!'
+puts '***Have fun!***'
+puts ''
+puts ''
+
+puts 'Player one, type in your name:'
+p1_name = gets.chomp
+name1 = valid_name(p1_name)
+X = 'X'.green
```
Comment (suretrust) [inline bin/main.rb]: When I tried running your code, I got the error below:
```ruby
bin/main.rb:78:in `<main>': undefined method `green' for "X":String (NoMethodError)
```
Screenshot 2020-03-21 at 03 22 00

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi team, 👋 

### Great job thus far. 🙌 Please see my comments. 👇 

- [ ] Your code could not run. It was complaining about the `green` method. Then I tried to remove it, I got the error below again:

```ruby
bin/main.rb:79:in `<main>': uninitialized constant Player (NameError)
```

**Please ensure to submit a code that runs properly. This will enable the reviewer to review your code and check if you have implemented the first reviewer's suggestions.** Kindly submit for another code review request once this has been fixed.  _Happy coding!_ 🎉

**Review by suretrust — APPROVED** [review]
Hi team, 👋 

### Great job implementing the suggested changes. 🙌 Let's get this merged! 👍 

_Good luck with your future milestones and happy coding!_ 🎉

**Pull request merged**